ecs/pod_creator.go

```diff
@@ -27,11 +31,227 @@ func NewBasicECSPodCreator(c cocoa.ECSClient, v cocoa.Vault) (*BasicECSPodCreato
+	if err != nil {
+		return nil, errors.Wrap(err, "translating the register task definition input to the correct type")
+	}
+
+	registerOut, err := m.client.RegisterTaskDefinition(ctx, taskDefinition)
+	if err != nil {
+		return nil, errors.Wrap(err, "registering task definition")
+	}
+
+	if registerOut.TaskDefinition == nil || registerOut.TaskDefinition.TaskDefinitionArn == nil {
+		return nil, errors.New("missing task definition")
+	}
+
+	taskDef := cocoa.NewECSTaskDefinition()
+	taskDef = taskDef.SetID(*registerOut.TaskDefinition.TaskDefinitionArn)
+	taskDef = taskDef.SetOwned(*utility.TruePtr())
+
+	runTask := &ecs.RunTaskInput{}
+	runTask.SetCluster(*mergedPodExecutionOpts.Cluster).
+		SetTaskDefinition(*taskDef.ID).
+		SetTags(registerOut.Tags).
+		SetEnableExecuteCommand(*mergedPodExecutionOpts.SupportsDebugMode).
+		SetPlacementStrategy(translateStrategy(mergedPodExecutionOpts.PlacementOpts.Strategy, mergedPodExecutionOpts.PlacementOpts.StrategyParameter))
+
+	runOut, err := m.client.RunTask(ctx, runTask)
+	if err != nil {
+		return nil, errors.Wrap(err, "running task")
+	}
+
+	if runOut.Failures != nil {
+		catcher := grip.NewBasicCatcher()
+		for _, failure := range runOut.Failures {
+			err = errors.Errorf("task '%s': %s: %s\n", *failure.Arn, *failure.Detail, *failure.Reason)
+			catcher.Add(err)
+		}
+		return nil, errors.Wrap(catcher.Resolve(), "running task")
+	}
+
+	if len(runOut.Tasks) == 0 || runOut.Tasks[0].TaskArn == nil {
+		return nil, errors.New("missing running task")
+	}
+
+	resources := cocoa.NewECSPodResources().
+		SetCluster(*mergedPodExecutionOpts.Cluster).
+		SetSecrets(allSecrets).
+		SetTaskDefinition(*taskDef).
+		SetTaskID(*runOut.Tasks[0].TaskArn)
+
+	options := NewBasicECSPodOptions().
+		SetClient(m.client).
+		SetVault(m.vault).
+		SetStatus(cocoa.Running).
+		SetResources(*resources)
+
+	p, err := NewBasicECSPod(options)
+
+	if err != nil {
+		return nil, errors.Wrap(err, "creating pod")
+	}
+
+	return p, nil
 }
 
 // CreatePodFromExistingDefinition creates a new pod backed by AWS ECS from an
 // existing definition.
 func (m *BasicECSPodCreator) CreatePodFromExistingDefinition(ctx context.Context, def cocoa.ECSTaskDefinition, opts ...*cocoa.ECSPodExecutionOptions) (cocoa.ECSPod, error) {
 	return nil, errors.New("TODO: implement")
 }
+
+// translateStringArrayToECSTagArray translates the strings into ECS tags.
+func translateStringArrayToECSTagArray(tags []string) []*ecs.Tag {
+	ecsTags := []*ecs.Tag{}
+	for _, tag := range tags {
+		ecsTag := &ecs.Tag{}
+		ecsTag.SetKey(tag)
+		ecsTags = append(ecsTags, ecsTag)
+	}
+	return ecsTags
+}
+
+// translateStrategy translates the strategy and parameter into ECS placement strategy.
+func translateStrategy(strategy *cocoa.ECSPlacementStrategy, param *cocoa.ECSStrategyParameter) []*ecs.PlacementStrategy {
+	placementStrat := ecs.PlacementStrategy{}
+	placementStrat.SetType(string(*strategy)).SetField(*param)
```

Comment:
This can panic. Use `utility.FromStringPtr`.

Reply:
For `strategy`, should I check for nil-ness as well since it is of type ` *cocoa.ECSPlacementStrategy`, and set it to be the empty string if the pointer is nil?

Reply:
That seems reasonable.